xds/src/main/java/io/grpc/xds/GcpAuthenticationFilter.java

```diff
@@ -119,17 +128,23 @@ public ClientInterceptor buildClientInterceptor(FilterConfig config,
       public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
           MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
 
-        /*String clusterName = callOptions.getOption(XdsAttributes.ATTR_CLUSTER_NAME);
-        if (clusterName == null) {
+        String clusterName = callOptions.getOption(CLUSTER_SELECTION_KEY);
+        if (clusterName.startsWith("cluster_specifier_plugin:")) {
           return next.newCall(method, callOptions);
-        }*/
+        }
 
-        // TODO: Fetch the CDS resource for the cluster.
-        // If the CDS resource is not available, fail the RPC with Status.UNAVAILABLE.
+        XdsConfig xdsConfig = callOptions.getOption(XDS_CONFIG_CALL_OPTION_KEY);
+        StatusOr<XdsClusterConfig> xdsCluster = xdsConfig.getClusters().get(clusterName);
+        CdsUpdate cdsUpdate = xdsCluster.getValue().getClusterResource();
```

Comment:
You can't safely call `StatusOr.getValue()` without first making sure the value is present.

Note that `getClusterResource()` is [guaranteed to not return null](https://github.com/grpc/grpc-java/blob/d60e6fc251d0c5c65b9e52e9ce1a84d3ec17c159/xds/src/main/java/io/grpc/xds/XdsConfig.java#L111).